solr/core/src/java/org/apache/solr/search/ExtendedDismaxQParser.java

```diff
@@ -199,6 +199,18 @@ public Query parse() throws SyntaxError {
       topQuery = FunctionScoreQuery.boostByValue(topQuery, boosts.get(0).asDoubleValuesSource());
     }
 
+    // If topQuery is a boolean query, unwrap the boolean query to check if it is just
```

Comment:
I recommend moving this to above the section right above here since FunctionScoreQuery may wrap the BooleanQuery or whatever `topQuery` happens to be.

Reply:
Addressed in 81ca321